plugins/cost-insights/src/hooks/useAlerts.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, {
+  useReducer,
+  Dispatch,
+  SetStateAction,
+  createContext,
+  useContext,
+  PropsWithChildren,
+} from 'react';
+import { Alert, Maybe } from '../types';
+
+export type AlertsContextProps = {
+  alerts: AlertState;
+  setAlerts: Dispatch<SetStateAction<Partial<AlertState>>>;
+};
+
+export const AlertsContext = createContext<AlertsContextProps | undefined>(
+  undefined,
+);
+
+export type AlertState = {
+  alerts: Alert[];
```

Comment:
alerts.alerts wasn't my favorite implementation and in places feels somewhat clunky but I wanted the alert context to reflect the data it represented, without using any redundant naming conventions e.g. alertState.alerts

Reply:
AlertState is defined within the hook, so no chance of users being confused since it isn't a public export (unless they're digging around for debugging purposes). 

In the code snippet above, alerts are being updated as a whole - the hook is called which returns a new set of alerts - which may or may not have a status. 

When alerts update, they are filtered by status and memoized at the page level and forwarded via props to consuming components. This ensures each component shares the same references and that they do not change between renders.